processor/ratelimitprocessor/gubernator.go

```diff
@@ -102,21 +117,41 @@ func (r *gubernatorRateLimiter) RateLimit(ctx context.Context, hits int) error {
 	})
 	if err != nil {
 		r.set.Logger.Error("error executing gubernator rate limit request", zap.Error(err))
+		r.requestTelemetry(ctx, []attribute.KeyValue{
+			telemetry.WithReason(telemetry.RequestErr),
+			telemetry.WithDecision("accepted"),
+		})
```

Comment:
So continuing my comment https://github.com/elastic/opentelemetry-collector-components/pull/562/files#r2098431295, maybe you could check `errors.Is(...,...)` and then set the metric in `processor.go`. This is so that `local.go` and `gubernator.go` won't need to have duplicated code.

Reply:
++, would be good to do as part of the followup.